CircuitPython_Pyloton/pyloton.py

```diff
@@ -0,0 +1,571 @@
+        self._hr_label = None
+        self._sp_label = None
+        self._cad_label = None
+        self._ams_label = None
+        self._hr_service = None
+        self._heart_y = None
+        self._speed_y = None
+        self._cad_y = None
+        self._ams_y = None
+
+        self.ams = None
+        self.cyc_connections = None
+        self.cyc_services = None
+
+        self.start = time.time()
+        self.track_artist = True
+
+        self.splash = displayio.Group(max_size=25)
+        self.loading_group = displayio.Group()
+
+        self._load_fonts()
+
+        self.sprite_sheet, self.palette = adafruit_imageload.load("/sprite_sheet.bmp",
+                                                                  bitmap=displayio.Bitmap,
+                                                                  palette=displayio.Palette)
+
+
+    def show_splash(self):
+        """
+        Shows the loading screen
+        """
+        if self.debug:
+            return
+        with open('blinka-pyloton.bmp', 'rb') as bitmap_file:
+            bitmap1 = displayio.OnDiskBitmap(bitmap_file)
+
+            tile_grid = displayio.TileGrid(bitmap1, pixel_shader=displayio.ColorConverter())
+
+
+            self.loading_group.append(tile_grid)
+
+            self.display.show(self.loading_group)
+
+            status_heading = label.Label(font=self.arial16, x=80, y=175,
+                                         text="Status", color=self.YELLOW)
+
+            rect = Rect(0, 165, 240, 75, fill=self.PURPLE)
+
+            self.loading_group.append(rect)
+            self.loading_group.append(status_heading)
+
+            self.display.show(self.loading_group)
+            time.sleep(.01)
+
+
+    def _load_fonts(self):
+        """
+        Loads fonts
+        """
+        self.arial12 = bitmap_font.load_font("/fonts/Arial-12.bdf")
+        self.arial16 = bitmap_font.load_font("/fonts/Arial-16.bdf")
+        self.arial24 = bitmap_font.load_font("/fonts/Arial-Bold-24.bdf")
+
+
+    def _status_update(self, message):
+        """
+        Displays status updates
+        """
+        if self.debug:
+            print(message)
+            return
+
+        if len(self.loading_group) == 3:
+            text_group = displayio.Group()
+            status = label.Label(font=self.arial12, x=10, y=200,
+                                 text='', color=self.YELLOW, max_glyphs=30)
+            status1 = label.Label(font=self.arial12, x=10, y=220,
+                                  text='', color=self.YELLOW, max_glyphs=30)
+
+            text_group.append(status)
+            text_group.append(status1)
+            self.loading_group.append(text_group)
+
+        if len(message) > 25:
+            status.text = message[:25]
+            status1.text = message[25:]
+
+            text_group.append(status)
+            text_group.append(status1)
+        else:
+            status.text = message
+            status1.text = ''
+            text_group.append(status)
```

Comment:
`status` and `status1` won't have a value if the `if len(...` is False. You need to make them instance vars to remember them.

Testing for `len(self.loading_group) == 3` is "brittle", because the `3` might change. You could test for `status` and/or `status` being non-`None` instead.

Reply:
Will do